libcxx/include/unordered_map

```diff
@@ -1024,6 +1029,15 @@ template <class _Key,
           class _Pred  = equal_to<_Key>,
           class _Alloc = allocator<pair<const _Key, _Tp> > >
 class _LIBCPP_TEMPLATE_VIS unordered_map {
+  // TODO(#106635): replace with _LIBCPP_CHECK_CONTAINER_VALUE_TYPE_REQUIREMENTS
+  // Remember to remove relevant headers when this is completed.
```

Comment:
I read the description of https://github.com/microsoft/STL/pull/3660 and it looks like the Standard does allow these types to be used. If that's the case, then https://github.com/llvm/llvm-project/issues/106635 should be closed until we have a LWG issue or a paper that changes the Standard, and these TODOs should be removed since we won't be able to address them.

Note that I have no love for reference types in these locations, but I also want to avoid adding non-actionable TODOs.

Reply:
I think MS/STL is allowing these as an extension? The linked wording from #106635 already prohibits `unordered_map<int, int&>`. We weren't catching it before, and have been Hyrum's Law'd.

Reply:
Okay, I get what's happening, thanks to cplusplus/draft#7249. I've parked #106635, and have replaced the TODOs with `// not an error`.

Reply:
Per https://github.com/llvm/llvm-project/issues/106635#issuecomment-2336761764, I've reopened the issue, and have reverted bb82c957a0e57a1d3d1d31ec71020f70ad659dad.

Reply:
So IIUC basically references in these locations are not permitted, and we should clean up the usage.

Reply:
References, `const`, `volatile`, etc. I'm especially leery of `map<int const, int>`, because it's in no way interface-distinguishable from `map<int, int>`, yet is its own instantiation.

I'm happy to take point on that over the next few releases, but we'll need to come up with a plan (and potentially coordinate with MS/STL and libstdc++).